src/hip_memory.cpp

```diff
@@ -1125,6 +1125,14 @@ hipError_t hipMemcpy(void* dst, const void* src, size_t sizeBytes, hipMemcpyKind
 
     hipError_t e = hipSuccess;
 
+    if(dst==NULL || dst==0 || src==NULL || src==0)
```

Comment:
HIP runtime is compiled using a C++ compiler. So comparison to both NULL as well as 0 does not make sense. Just comparing to NULL is sufficient.